### DISCUSSION TITLE
Filter Greater Than Or Less Than Value

### DISCUSSION THREAD

**Discussion by mr1716** [netbox-community/netbox#14970] [Ideas]
For certain items such as Manufacturers or Device Roles where you have an integer field, it would be great to have a filter available that would be like:
Manufacturers where Device Types > 1
Device Roles where # of Devices <3

**Comment by LuPo** [comment]
You can use lookup expressions (separated with a double underscore) as described [here](https://docs.netbox.dev/en/stable/reference/filtering/#lookup-expressions) on __specific__ model fields:
`/ipam/vlans/?vid__gt=900`
It is not possible, however, to apply them while filtering on data descriptors.
This means that you can query for a device with a specific role id: `/dcim/devices/?role_id=15`, but you can not search for devices with role id smaller than any given integer like in this example: `/dcim/devices/?role_id__lte=15`

There is also still an open issue, #7604, to support this in the UI on the object filter tab.

**Reply by candlerb** [reply]
You could do that with `/dcim/devices/?role_id=1&role_id=2&role_id=3&...&role_id=15`

However, I don't think that's what the OP asked. They wanted attributes which represent *counts* of related objects, e.g. to filter on manufacturers where the *number of* child device types is greater than N, or device roles where the *number of* devices with that role is less than M.

The web UI has table columns for these counts (and you can sort on those columns):

```
class ManufacturerTable(ContactsColumnMixin, NetBoxTable):
    name = tables.Column(
        verbose_name=_('Name'),
        linkify=True
    )
    devicetype_count = columns.LinkedCountColumn(
        viewname='dcim:devicetype_list',
        url_params={'manufacturer_id': 'pk'},
        verbose_name=_('Device Types')
    )
```

Is this also available in the REST API? Yes:

```
class ManufacturerSerializer(NetBoxModelSerializer):
    url = serializers.HyperlinkedIdentityField(view_name='dcim-api:manufacturer-detail')
    devicetype_count = serializers.IntegerField(read_only=True)
```

```
# curl -sS -H "Authorization: Token $TOKEN" "$NETBOX/api/dcim/manufacturers/?limit=1" | jq .
{
  "count": 14,
  "next": "https://netbox.example.net/api/dcim/manufacturers/?limit=1&offset=1",
  "previous": null,
  "results": [
    {
      "id": 30,
      "url": "https://netbox.example.net/api/dcim/manufacturers/30/",
      "display": "APC",
      "name": "APC",
      "slug": "apc",
      "description": "",
      "tags": [],
      "custom_fields": {},
      "created": "2024-02-16T21:04:12.799007Z",
      "last_updated": "2024-02-16T21:04:12.799070Z",
      "devicetype_count": 0,    <<<<
      "inventoryitem_count": 0,    <<<<
      "platform_count": 0    <<<<
    }
  ]
}
```

Can we filter on these? Unfortunately not.

```
# curl -sS -H "Authorization: Token $TOKEN" "$NETBOX/api/dcim/manufacturers/?devicetype_count=0" |
    jq -r '.results[]|[.id,.devicetype_count]|@csv'
30,0
20,2
29,1
2,0
11,0
17,1
21,1
4,0
9,3
15,0
12,2
10,1
18,1
16,2
```

Right now, I think your best solution is to query for all objects and filter them client-side

```
# curl -sS -H "Authorization: Token $TOKEN" "$NETBOX/api/dcim/manufacturers/" |
    jq -r '.results[]|select(.devicetype_count==0)|[.id,.devicetype_count]|@csv'
30,0
2,0
11,0
4,0
15,0
```

**Reply by LuPo** [reply]
Thank you @candlerb for your explanation. I see now where I missed the point.
